man/faq/developer/howto-coercion-data-frame.Rmd

```diff
@@ -112,22 +112,35 @@ vec_cbind(data.frame(x = 1:3), data.table(y = "foo"))
 
 ## A tibble example
 
-In this example we implement coercion methods for a tibble subclass:
+In this example we implement coercion methods for a tibble subclass that carries a colour as metadata:
 
 ```{r}
 # User constructor
-my_tibble <- function(...) {
-  new_my_tibble(tibble::tibble(...))
+my_tibble <- function(colour = NULL, ...) {
+  new_my_tibble(tibble::tibble(...), colour = colour)
 }
 # Developer constructor
-new_my_tibble <- function(x) {
+new_my_tibble <- function(x, colour = NULL) {
   stopifnot(is.data.frame(x))
-  tibble::new_tibble(x, class = "my_tibble", nrow = nrow(x))
+  tibble::new_tibble(
+    x,
+    colour = colour,
+    class = "my_tibble",
+    nrow = nrow(x)
+  )
+}
+
+df_colour <- function(x) {
+  if (inherits(x, "my_tibble")) {
```

Comment:
Do we need this for a simple example?

Reply:
I think so, it's used later in the implementations. It also implies that there is no additional behaviour for colour tibbles, in the sense that the colour interface is defined on data frames as well.